packages/jest-jasmine2/src/setup-jest-globals.js

```diff
@@ -48,59 +46,16 @@ const addSuppressedErrors = result => {
   }
 };
 
-function addAssertionErrors(result) {
-  const {
-    assertionCalls,
-    expectedAssertionsNumber,
-    isExpectingAssertions,
-  } = getState();
-  setState({
-    assertionCalls: 0,
-    expectedAssertionsNumber: null,
-  });
-  if (
-    typeof expectedAssertionsNumber === 'number' &&
-    assertionCalls !== expectedAssertionsNumber
-  ) {
-    const expected = EXPECTED_COLOR(
-      pluralize('assertion', expectedAssertionsNumber),
-    );
-    const message = new Error(
-      matcherHint('.assertions', '', String(expectedAssertionsNumber), {
-        isDirectExpectCall: true,
-      }) +
-        '\n\n' +
-        `Expected ${expected} to be called but only received ` +
-        RECEIVED_COLOR(pluralize('assertion call', assertionCalls || 0)) +
-        '.',
-    ).stack;
+const addAssertionErrors = result => {
+  const assertionErrors = extractExpectedAssertionsErrors();
+  if (assertionErrors.length) {
     result.status = 'failed';
-    result.failedExpectations.push({
-      actual: assertionCalls,
-      expected: expectedAssertionsNumber,
-      message,
-      passed: false,
-    });
-  }
-  if (isExpectingAssertions && assertionCalls === 0) {
-    const expected = EXPECTED_COLOR('at least one assertion');
-    const received = RECEIVED_COLOR('received none');
-    const message = new Error(
-      matcherHint('.hasAssertions', '', '', {
-        isDirectExpectCall: true,
-      }) +
-        '\n\n' +
-        `Expected ${expected} to be called but ${received}.`,
-    ).stack;
-    result.status = 'failed';
-    result.failedExpectations.push({
-      actual: 'none',
-      expected: 'at least one',
-      message,
+    result.failedExpectations = assertionErrors.map(error => ({
```

Comment:
@aaronabramov ah good point. i also found extractExpectedAssertionsErrors()'s return type is Array<Error>
and Error does not has assertionCall and expectedAssertionNumber properties. if i added them to Error it won't pass the flow check.
what do you think change extractExpectedAssertionsErrors to return Array<Object> Where Object is {
    actual: assertionCalls,
    expected: expectedAssertionsNumber
    message: message,
    passed: false,
  };
Then we don;t have to loop through the error to create the object. we can just concat two arrays. it seems it also better for performance.